codice/ddf
# DDF-4106 Starts migration of Admin UI to React

#### What does this PR do?
Adds the ability to use React in the Admin UI.  

As a proof of concept.
![image](https://user-images.githubusercontent.com/11984853/49605447-bdfbe480-f94d-11e8-8f9b-9cd124e8ad69.png)
I converted the configuration tab to use React entirely, until the point at which the editing / create modal is opened.  

I also updated the logviewer section to pull in the logviewer directly rather than iframe it in to give users a better experience.  We could do something similar for other applications.

There were two long standing bugs that were fixed as part of this PR as well.  One is that the Admin UI used to scroll back to the top of the page whenever the remove confirmation was dismissed (for configurations).  The other is that the Admin UI wouldn't update correctly when the final configuration in a service factory was deleted.  It would stick around even though the backend updated correctly and removed the final configuration.  

#### Who is reviewing this PR?
@codice/ui 
@djblue 
@rzwiefel
@tbatie 
@adimka 

#### How should this be tested?
Go to the Admin UI and verify everything works as expected.  Add a configuration, remove a configuration.  Go to the logviewer and verify it works.  Verify the logviewer can't be opened as a separate app anymore.  

There were two long standing bugs that were fixed as part of this PR.  One is that the Admin UI used to scroll back to the top of the page whenever the remove confirmation was dismissed (for configurations).  The other is that the Admin UI wouldn't update correctly when the final configuration in a service factory was deleted.  It would stick around even though the backend updated correctly and removed the final configuration.  

#### Any background context you want to provide?
This will position us to start pulling in React apps (such as the logviewer, which this PR does) as first class supported components in the Admin UI.  It'll also allow position us to start improving the Admin UI in general.

#### What are the relevant tickets?
[DDF-4106](https://codice.atlassian.net/browse/DDF-4106)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
